Framework/Crystal/src/CountReflections.cpp

```diff
@@ -0,0 +1,217 @@
+#include "MantidCrystal/CountReflections.h"
+#include "MantidCrystal/PeakStatisticsTools.h"
+
+#include "MantidAPI/Sample.h"
+#include "MantidAPI/WorkspaceProperty.h"
+
+#include "MantidDataObjects/PeaksWorkspace.h"
+
+#include "MantidGeometry/Crystal/PointGroupFactory.h"
+#include "MantidGeometry/Crystal/ReflectionCondition.h"
+#include "MantidGeometry/Crystal/OrientedLattice.h"
+
+#include "MantidKernel/make_unique.h"
+#include "MantidKernel/ListValidator.h"
+
+namespace Mantid {
+namespace Crystal {
+
+using Mantid::Kernel::Direction;
+
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CountReflections)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string CountReflections::name() const { return "CountReflections"; }
+
+/// Algorithm's version for identification. @see Algorithm::version
+int CountReflections::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string CountReflections::category() const {
+  return "Crystal\\Peaks";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string CountReflections::summary() const {
+  return "Calculates statistics for a PeaksWorkspace based on symmetry and "
+         "counting reflections.";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void CountReflections::init() {
+  declareProperty(Kernel::make_unique<WorkspaceProperty<PeaksWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "A workspace with peaks to calculate statistics for. Sample "
+                  "with valid UB-matrix is required.");
+
+  auto centeringSymbols = getAllReflectionConditionSymbols();
+  declareProperty("LatticeCentering", centeringSymbols[0],
```

Comment:
Another option would be dedicated property types which accept both forms (one for each of centering, point group, space group). That would definitely be better for backwards compatibility. It would also eliminate the need to do this "get a list of all x, make a validator, check what string the user supplied, create object accordingly", which is found anywhere point group and/or lattice centering are used.